### PR TITLE
Node v18, Webdriver v8

### PR DESCRIPTION
**Pull request by icio**
Upgrade webdriverio from v4 to v8 and node from v10 to v18. The primary change is that webdriverio dropped their synchronous API and you now have to `await` everything, but with this I'm also specifying that we should use Selenium v4 by default. Somewhere in that upgrade we've lost the ability to paste into and resize iOS devices from spec tests, which is probably fine (the unit tests check this too).

### CODE REVIEW

**Comment by adamroyjones** [comment]
> you now have to `await` everything

Ah, function colouring, ain't it wonderful.

File: .cloudbuild/e2e.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # yaml-language-server: $schema=https://json.schemastore.org/cloudbuild.json
 substitutions:
-  _NODE10: node:10.24.1-slim
+  _NODE: node:18-slim
```
Comment (adamroyjones) [inline .cloudbuild/e2e.yaml]: How come we use Alpine in one context and Debian here?

File: test/common.spec.mjs [hunk 2]
```diff
@@ -76,8 +80,15 @@ function hasURL(substr) {
  * @param {string} selector
  * @returns {NavTest}
  */
-function hasElement(selector) {
-  return browser => browser.$(selector).isExisting();
+export function hasElement(selector) {
+  return async function hasElementTest(browser) {
+    try {
+      const e = await browser.$(selector);
```
Comment (adamroyjones) [inline test/common.spec.mjs:86]: (I suddenly feel old.)

File: test/server.mjs [hunk 3]
```diff
@@ -219,18 +227,17 @@ async function expectNoError(api, key) {
...
-    launchProxy(a).then(url => console.log('🚆', url));
-  } else {
-    // Launch on the given port.
-    const port = parseInt(process.argv[2], 10);
-    if (isNaN(port)) throw new Error('invalid port');
-    a.listen(port, 'localhost');
-    console.log('http://localhost:' + port);
-  }
-}
+// TODO https://stackoverflow.com/questions/57838022/detect-whether-es-module-is-run-from-command-line-in-node
```
Comment (adamroyjones) [inline test/server.mjs]: Is this still needed?

File: test/wdio.8.mjs [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+        'bstack:options': {
+          'os': 'Windows',
+          'osVersion': '7',
+          'sendKeys': true,
+          'seleniumVersion': '3.5.2',
+        },
+      },
+
+      // Android
+      // Reminder set for 16-01-2023 to uncomment this next block and check if
```
Comment (adamroyjones) [inline test/wdio.8.mjs]: (This date is in the past!)

File: test/wdio.8.mjs [hunk 5]
```diff
@@ -0,0 +1,463 @@
...
+      {
+        'browserName': 'Safari',
+        'browserVersion': '13.0',
+        'bstack:options': {
+          'os': 'OS X',
+          'osVersion': 'Catalina',
+        },
+      },
+      // Finding that High Sierra isn't starting properly:
+      // https://automate.browserstack.com/dashboard/v2/builds/ffecf909566a3dd424c71404338350fe6ccecabf/sessions/8fb5ed210e940700b992559292c8516ca3fc33e3
```
Comment (adamroyjones) [inline test/wdio.8.mjs]: Do we want a TODO here with a reminder?

File: test/wdio.8.mjs [hunk 6]
```diff
@@ -0,0 +1,463 @@
...
+          'osVersion': '7.0',
+          'deviceName': 'Samsung Galaxy S8',
+          'realMobile': 'true',
+        },
+        'browserName': 'Android'
+      },
+
+      // iOS
+      // iOS11 currently failing to start with 'COULD NOT START MOBILE BROWSER'.
+      // https://automate.browserstack.com/dashboard/v2/builds/2840b7364373dc5f6f27be58d7bb33374d492668?overallStatus=error
```
Comment (adamroyjones) [inline test/wdio.8.mjs]: Do we want a TODO here with a reminder?

**Review by adamroyjones — COMMENTED** [review]
A few minor questions, but it looks like another heroic effort.

File: .cloudbuild/e2e.yaml (hunk 1 above)
Comment (icio) [reply .cloudbuild/e2e.yaml]: Ah, because I only tested one! We need `ps` available which only the alpine flavour comes with.

Comment (adamroyjones) [reply .cloudbuild/e2e.yaml]: ! You tell the truth.

File: test/server.mjs (hunk 3 above)
Comment (icio) [reply test/server.mjs]: Reinstated in 73077d09d1b4ca66c596a611006e300d19f607d3

File: .cloudbuild/e2e.yaml (hunk 1 above)
Comment (icio) [reply .cloudbuild/e2e.yaml]: Changed in dc2d0a17e5b32dcfe0f934f088f78741fab35bd0.

File: test/wdio.8.mjs (hunk 4 above)
Comment (icio) [reply test/wdio.8.mjs]: I'm going to tidy up these capabilities in another pass.

**Review by adamroyjones — APPROVED** [review]
(As you mentioned that there'll be another pass with the capabilities, all's well to me. Thanks for doing this—I can hardly imagine the tedium of colouring everything red. Or green.)

**Pull request merged**